api/internal/handler/data_loader/route_export.go

```diff
@@ -0,0 +1,441 @@
+				}
+				return nil, err
+			}
+
+			_service := service.(*entity.Service)
+			servicePlugins = _service.Plugins
+			serviceLabels = _service.Labels
+		}
+
+		if route.Upstream != nil {
+			extensions["x-apisix-upstream"] = route.Upstream
+		} else if route.UpstreamID != nil && route.Upstream == nil {
+			upstreamID := utils.InterfaceToString(route.UpstreamID)
+			upstream, err := h.upstreamStore.Get(context.Background(), upstreamID)
+			if err != nil {
+				if err == data.ErrNotFound {
+					return nil, fmt.Errorf("upstream id: %s not found", route.UpstreamID)
+				}
+				return nil, err
+			}
+			extensions["x-apisix-upstream"] = upstream
+		} else if route.UpstreamID == nil && route.Upstream == nil && route.ServiceID != nil {
+			_service := service.(*entity.Service)
+			if _service.Upstream != nil {
+				extensions["x-apisix-upstream"] = _service.Upstream
+			} else if _service.Upstream == nil && _service.UpstreamID != nil {
+				upstreamID := utils.InterfaceToString(_service.UpstreamID)
+				upstream, err := h.upstreamStore.Get(context.Background(), upstreamID)
+				if err != nil {
+					if err == data.ErrNotFound {
+						return nil, fmt.Errorf("upstream id: %s not found", _service.UpstreamID)
+					}
+					return nil, err
+				}
+				extensions["x-apisix-upstream"] = upstream
+			}
+		}
+
+		if route.Host != "" {
+			extensions["x-apisix-host"] = route.Host
+		}
+
+		if route.Hosts != nil {
+			extensions["x-apisix-hosts"] = route.Hosts
+		}
+
+		//Parse Labels
+		labels, err = parseLabels(route, serviceLabels, labels)
+		if err != nil {
+			log.Errorf("parseLabels err: ", err)
+			return nil, err
+		}
+
+		if labels != nil {
+			extensions["x-apisix-labels"] = labels
+		}
+
+		if route.RemoteAddr != "" {
+			extensions["x-apisix-remote_addr"] = route.RemoteAddr
+		}
+
+		if route.RemoteAddrs != nil {
+			extensions["x-apisix-remote_addrs"] = route.RemoteAddrs
+		}
+
+		if route.FilterFunc != "" {
+			extensions["x-apisix-filter_func"] = route.FilterFunc
+		}
+
+		if route.Script != nil {
+			extensions["x-apisix-script"] = route.Script
+		}
+
+		if route.ServiceProtocol != "" {
+			extensions["x-apisix-service_protocol"] = route.ServiceProtocol
+		}
+		if route.Vars != nil {
+			extensions["x-apisix-vars"] = route.Vars
+		}
+
+		// analysis route.URIs
+		routeURIs := []string{}
+		if route.URI != "" {
+			routeURIs = append(routeURIs, route.URI)
+		}
+
+		if route.Uris != nil {
+			routeURIs = route.Uris
+		}
+
+		for _, uri := range routeURIs {
+			if strings.Contains(uri, "*") {
+				paths[strings.Split(uri, "*")[0]+"{params}"] = pathItem
+				// add params introduce
+				paramsRefs = append(paramsRefs, &openapi3.ParameterRef{
+					Value: &openapi3.Parameter{
+						In:          "path",
+						Name:        "params",
+						Required:    true,
+						Description: "params in path",
+						Schema:      &openapi3.SchemaRef{Value: &openapi3.Schema{Type: "string"}}}})
+			} else {
+				paths[uri] = pathItem
+			}
+		}
```

Comment:
Can we split `analysis  route.URIs` to another function?

Reply:
No problem, but there are some problems that need to be solved by voting. I'll deal with them after the results come out

Reply:
it seems that this case has been fixed, right @Jaycean ?

Reply:
done